rotkehlchen/db/snapshots.py

```diff
@@ -0,0 +1,140 @@
+        self.db = db_handler
+
+    def get_timed_balances(self, timestamp: Timestamp) -> List[DBAssetBalance]:
+        """Retrieves the timed_balances from the db for a given timestamp"""
+        balances_data = []
+        cursor = self.db.conn.cursor()
+        timed_balances_result = cursor.execute(
+            'SELECT category, time, amount, currency, usd_value FROM timed_balances '
+            'WHERE time=?', (timestamp,),
+        )
+        for data in timed_balances_result:
+            balances_data.append(
+                DBAssetBalance(
+                    category=BalanceType.deserialize_from_db(data[0]),
+                    time=data[1],
+                    amount=data[2],
+                    asset=Asset(data[3]),
+                    usd_value=data[4],
+                ),
+            )
+        return balances_data
+
+    def get_timed_location_data(self, timestamp: Timestamp) -> List[LocationData]:
+        """Retrieves the timed_location_data from the db for a given timestamp"""
+        location_data = []
+        cursor = self.db.conn.cursor()
+        timed_location_data = cursor.execute(
+            'SELECT time, location, usd_value FROM timed_location_data '
+            'WHERE time=?',
+            (timestamp,),
+        )
+        for data in timed_location_data:
+            location_data.append(
+                LocationData(
+                    time=data[0],
+                    location=data[1],
+                    usd_value=data[2],
+                ),
+            )
+        return location_data
+
+    def create_zip(
+        self,
+        timed_balances: List[DBAssetBalance],
+        timed_location_data: List[LocationData],
+    ) -> Tuple[bool, str]:
+        """Creates a zip file of csv files containing timed_balances and timed_location_data."""
+        dirpath = Path(mkdtemp())
+        success, msg = self._export(
+            timed_balances=timed_balances,
+            timed_location_data=timed_location_data,
+            directory=dirpath,
+        )
+        if not success:
+            return False, msg
+
+        files: List[Tuple[Path, str]] = [
+            (dirpath / BALANCES_FILENAME, BALANCES_FILENAME),
+            (dirpath / LOCATION_DATA_FILENAME, LOCATION_DATA_FILENAME),
+        ]
+        with ZipFile(file=dirpath / 'snapshot.zip', mode='w', compression=ZIP_DEFLATED) as archive:
+            for path, filename in files:
+                if not path.exists():
+                    continue
+
+                archive.write(path, filename)
+                path.unlink()
+
+        success = False
+        filename = ''
+        if archive.filename is not None:
+            success = True
+            filename = archive.filename
+
+        return success, filename
+
+    def export(self, timestamp: Timestamp, directory_path: Optional[Path]) -> Tuple[bool, str]:
+        """Export the database's snapshot for specified timestamp."""
```

Comment:
Add a bit more info in the docstring here, explaining the difference in behavior between providing the directory_path and not.